include/openthread/dataset.h

```diff
@@ -125,14 +125,37 @@ struct otPskc
  */
 typedef struct otPskc otPskc;
 
+#define OT_THREAD_1_1_SECURITY_POLICY_FLAGS_LENGTH 1 ///< The Thread 1.1 Security Policy Flags length.
+#define OT_THREAD_1_2_SECURITY_POLICY_FLAGS_LENGTH 2 ///< The Thread 1.2 Security Policy Flags length.
+
 /**
  * This structure represent Security Policy.
  *
  */
 typedef struct otSecurityPolicy
 {
-    uint16_t mRotationTime; ///< The value for thrKeyRotation in units of hours
-    uint8_t  mFlags;        ///< Flags as defined in Thread 1.1 Section 8.10.1.15
+    /*
+     * The value for thrKeyRotation in units of hours.
+     *
+     */
+    uint16_t mRotationTime;
+
+    /*
+     * The Security Policy Flags as defined in Thread 1.2 Section 8.10.1.15.
+     * For Thread 1.1 or lower, only the first byte is available. For Thread
+     * 1.2 or higher, the available `mFlags` bytes are indicated by `mFlagsLength`.
+     *
+     */
+    uint8_t mFlags[OT_THREAD_1_2_SECURITY_POLICY_FLAGS_LENGTH];
+
+    /**
+     * The length of the Security Policy Flags (`mFlags`).
+     *
+     * @sa OT_THREAD_1_1_SECURITY_POLICY_FLAGS_LENGTH
+     * @sa OT_THREAD_1_2_SECURITY_POLICY_FLAGS_LENGTH
+     *
+     */
+    uint8_t mFlagsLength;
```

Comment:
Can we remove this field so that we always store 2B flags in the Dataset, even for 1.1 devices. We can set the 2nd byte to a default value for 1.1 SecurityPolicyTlv.

Reply:
If remove this field, a 1.2 device will always store 2-bytes security policy flags when receiving an operational dataset from a Thead 1.1 device. I don't think this is what we want.

Reply:
@librasungirl @abtink Thoughts?